Protos/V1/agent_common.proto

```diff
@@ -16,14 +18,21 @@ message CreateTaskRequest {
     DataChunk task_payload = 3;
   }
 
-  string communication_token = 4;
+  string communication_token = 4; /** Communication token received by the worker during task processing */
 }
 
 message CreateTaskReply {
+  message ResultIdName {
+    string name = 1; /** Result given name. */
+    string result_id = 2; /** Result Unique ID. */
+  }
```

Comment:
Why not use a map ? https://protobuf.dev/programming-guides/encoding/#maps

Reply:
finally, we don't need this message